enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+  want to deploy a cluster using my own automation for provisioning.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to deploy a cluster using my own automation for provisioning.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to interactively deploy a cluster and receive immediate feedback about
+  configuration errors and problems.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to deploy a cluster to a vSphere where I have permissions only to set
+  the boot image for one or more existing VMs.
+- As a user in a disconnected environment with no existing management cluster
+  and strict security protocols, I  want to deploy a cluster using only
+  physical media and access to the host consoles via a KVM switch.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+Further details will be discussed in future enhancements.
+
+### Risks and Mitigations
+
+The agent-based installer adds yet *another* method of installing OpenShift,
+and overlaps some other installation methods in applicability. Thus, in some
+cases it will contribute to even more decisions the user has to make before
+getting started.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Should the command-line tool be a subcommand of openshift-installer, or a
+  standalone binary?
+- Will the containers for the assisted components be retrieved from the same
+  release payload, or will we require the user to do additional mirroring (e.g.
+  of MCE) as well?
+
+### Test Plan
+
+The agent-based installer will be covered by end-to-end testing using virtual
+machines (in a baremetal-like configuration), automated by some variation on
+the metal platform
+[dev-scripts](https://github.com/openshift-metal3/dev-scripts/#readme). This is
+similar to the testing employed on both the baremetal IPI and assisted
+installation flows.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
```

Comment:
This section could talk about the goal of supporting versionable manifests as inputs so that the API of the image generator follows our existing support and deprecation guarantees for anyone integrating the tool with their workflow.

Reply:
Let's leave this for the detailed enhancement to be written next.